File: examples/csharp/micro/UserService.cs
using System.Collections.Generic;

namespace Micro
{
    public class UserService
    {
        private readonly AuthService _authService = new AuthService();

        public Dictionary<string, string> GetUser(string token)
        {
            var userId = _authService.VerifyToken(token);
            return new Dictionary<string, string>
            {
                { "id", userId },
                { "token", token }
            };
        }

        public string UpdateUser(string token, string action)
        {
            var userId = _authService.VerifyToken(token);
            if (userId != null)
            {
                _authService.RunAdminCommand(userId, action);
            }
            return userId;
        }
    }
}

File: examples/cpp/micro/user_service.cpp
#include "user_service.h"
#include "auth_service.h"

namespace micro {

UserInfo get_user(const std::string& token) {
    std::string user_id = verify_token(token);
    return UserInfo{user_id, token};
}

std::string update_user(const std::string& token, const std::string& action) {
    std::string user_id = verify_token(token);
    if (!user_id.empty()) {
        run_admin_command(user_id, action);
    }
    return user_id;
}

}
